webnode/src/redux/epics/node-epics.js

```diff
@@ -21,8 +21,12 @@ const registerWebnodeEpic = (action$, store) => {
   return action$
     .ofType(nodeActions.NODE_INITIALIZE, nodeActions.NODE_RESET)
     .mergeMap(action => {
-      const { node } = store.getState();
-      return Observable.fromPromise(brokerNode.registerWebnode(node.id))
+      const { id } = store.getState().node;
+      const brokerNodeUrl = nodeSelectors.brokerNodeUrl(store.getState());
```

Comment:
Selectors are typically used for derived state in the views. This introduces a race condition since you end up calling `getState` in an epic. The selected node should be passed in as part of the payload.

We can work on cleaning this up later though since it's higher priority to get this working e2e.

Reply:
k